engine/src/lib.rs

```diff
@@ -345,6 +345,7 @@ impl IncludeCppEngine {
             .clang_args(make_clang_args(inc_dirs, extra_clang_args))
             .derive_copy(false)
             .derive_debug(false)
+            .derive_default(true) // Needed to safely construct POD structs with bitfields
```

Comment:
Let me know if you think this'll have detrimental code size or correctness effects elsewhere. As per [bindgen's documentation](https://rust-lang.github.io/rust-bindgen/using-bitfields.html), it's the only ergonomic way to construct types with bitfields, but I can leave it out if it breaks other things.